app/repositories/OptOutSessionDataRepository.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import cats.data.OptionT
+import models.incomeSourceDetails.{TaxYear, UIJourneySessionData}
+import models.itsaStatus.ITSAStatus
+import models.itsaStatus.ITSAStatus.ITSAStatus
+import models.optout.OptOutSessionData
+import play.api.libs.json.{Json, OFormat}
+import repositories.OptOutContextData.{statusToString, stringToStatus}
+import services.optout.OptOutProposition
+import services.optout.OptOutProposition.createOptOutProposition
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.OptOutJourney
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class OptOutSessionDataRepository @Inject()(val repository: UIJourneySessionDataRepository) {
+
+  def saveIntent(intent: TaxYear)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Boolean] = {
+    OptionT(repository.get(hc.sessionId.get.value, OptOutJourney.Name)).
```

Comment:
should it not add optOutSessionData if missing?

Reply:
That's a good question. The only way that the user should get to this point is by going through the journey, and by doing that this will be setup at the start, so, under normal circumstances, it should never be missing.

I suppose some user could jump directly to this page, but that would show the error page (correctly, I think) as it would not be able to pull back the `OptOutProposition` if the session data were missing.

The only other scenario I can think of is if the user pauses between the setup at the start, and coming to this page. The service wipes old sessions from mongo, but the user would also be logged-out and having to sign-in again, so they would have to re-start the journey.

Are there other scenarios you can think of that might be problematic?

I think that it's simple in it's current form and covers all of the scenarios I can think of. If we don't need to do this for a particular reason, then I'd rather keep this as it is.